includes/services/extended/nextdoor.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/**
+ * Nextdoor service definition.
+ */
+
+class Keyring_Service_Nextdoor extends Keyring_Service_OAuth2 {
+	const NAME  = 'nextdoor';
+	const LABEL = 'Nextdoor';
+
+
+	function __construct() {
+		parent::__construct();
+
+		// Enable "basic" UI for entering key/secret
+		if ( ! KEYRING__HEADLESS_MODE ) {
+			add_action( 'keyring_nextdoor_manage_ui', array( $this, 'basic_ui' ) );
+			add_filter( 'keyring_nextdoor_basic_ui_intro', array( $this, 'basic_ui_intro' ) );
+		}
+
+		$this->set_endpoint("access_token", "https://auth.nextdoor.com/v2/token", "POST" );
+        $this->set_endpoint("authorize", "https://nextdoor.com/v3/authorize/?scope=openid%20post:write&", "POST");
+
+		$creds = $this->get_credentials();
+		if ( is_array( $creds ) ) {
+			$this->app_id = $creds['app_id'];
+			$this->key    = $creds['key'];
+			$this->secret = $creds['secret'];
+		}
+		$admin_url          = Keyring_Util::admin_url();
+		$this->redirect_uri = substr( $admin_url, 0, strpos( $admin_url, '?' ) );
+
+		// Send authorization via header
+		$this->authorization_header = 'Bearer';
+		add_filter( 'keyring_nextdoor_request_token_params', array( $this, 'request_token_params' ) );
+		add_filter( 'keyring_nextdoor_verify_token_params', array( $this, 'verify_token_params' ) );
+		add_filter( 'keyring_nextdoor_verify_token_post_params', array( $this, 'verify_token_post_params' ) );
+		add_filter( 'keyring_access_token', array( $this, 'fix_access_token_meta' ), 10, 2 );
+	}
+
+	function basic_ui_intro() {
+		/* translators: url */
+		echo '<p>' . sprintf( __( 'To use the Nextdoor service you need to be manually approved. Please reach out to get your client id and client secret.', 'keyring' )) . '</p>';
```

Comment:
The end user doesn't need to provide client id and secret, which is stored in the WordPress server.

Reply:
Anybody using this service will need a client ID and secret. We should probably provide a link to a support doc or something where they can request credentials.

Reply:
We don't have a full on support doc yet and I'm not sure if we will be creating one soon. Do you think leaving an email to contact would be fine?